src/Types/JsonWsp.purs

```diff
@@ -297,3 +309,29 @@ parseValue outer = do
   coins <- parseFieldToBigInt o "coins" <|> Left (TypeMismatch "Expected 'coins' to be an Int or a BigInt")
   Assets assetsMap <- fromMaybe (Assets Map.empty) <$> getFieldOptional o "assets"
   pure $ mkValue (wrap coins) (wrap assetsMap)
+
+-- chain tip query
+
+type ChainTipQR = ChainOrigin |+| ChainPoint
```

Comment:
Is there a reason to use `Untagged.Union` instead of a sum type?

Reply:
I thought it might be easier for users to define this as a sum type, but either way works I suppose

Reply:
In general, I'm in favour of using Variant for such purposes, but it is a little bit more vocal. I will change it to a sum type.